Realm/Realm/Linq/SortClauseVisitor.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace Realms
+{
+    internal class SortClauseVisitor
+    {
+        public OrderingClause _orderingClause;
+
+        public OrderingClause VisitOrderClause(MethodCallExpression orderClause)
+        {
+            _orderingClause = new OrderingClause();
+
+            if (orderClause.Method.Name == nameof(Queryable.OrderBy)
+                || orderClause.Method.Name == nameof(Queryable.ThenBy))
+            {
+                _orderingClause.IsAscending = true;
+            }
+            else if (orderClause.Method.Name == nameof(Queryable.OrderByDescending)
+                || orderClause.Method.Name == nameof(Queryable.ThenByDescending))
+            {
+                _orderingClause.IsAscending = false;
+            }
+            else
+            {
+                throw new NotSupportedException(orderClause.Method.Name + " is not a supported ordering method");
+            }
+
+            var lambda = (LambdaExpression)RealmResultsVisitor2.StripQuotes(orderClause.Arguments[1]);
```

Comment:
I am not a fan of this. I don't think that `StripQuotes` is in the right place at the moment.